frontend/src/metabase-types/api/entity-id.ts

```diff
@@ -0,0 +1,24 @@
+const NANOID_ALPHABET =
+  "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const NANOID_LENGTH = 21;
+
+declare const __brand: unique symbol;
+type Brand<T, B> = T & { readonly [__brand]: B };
+type NanoID = Brand<string, "NanoID">;
+
+export const isNanoID = (id: unknown): id is NanoID => {
+  return (
+    typeof id === "string" &&
+    id.length === NANOID_LENGTH &&
+    new RegExp(`^[${NANOID_ALPHABET}]+$`).test(id)
```

Comment:
you could combine these lines to
```suggestion
    new RegExp(`^[${NANOID_ALPHABET}]{${NANOID_LENGTH}}$`).test(id)
```

Idk if the above syntax is actually correct or not, but I want to result to be something like `/^[_-abc...]{21}$/`

Reply:
Is that necessarily better? I kinda like the obvious explicit check to the length - it's easier to read

Reply:
not necessarily better I'd say. It's just that I'm so used to use Reg ex to also check the string length, so I thought that might be a bit more streamline, but in this case the number isn't hard-coded, and using a variable makes the regex construction looks too complicated. Feels free to leave it as is.